aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -264,15 +265,25 @@ private boolean isGlueIcebergTable(Table table) {
 
   @Override
   public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    TableOperations ops = newTableOps(identifier);
+    TableMetadata lastMetadata;
+    try {
+      lastMetadata = ops.current();
+    } catch (NoSuchKeyException ex) {
+      // S3 files already deleted, can not get S3 object through metadata location
+      lastMetadata = null;
+    }
     try {
-      TableOperations ops = newTableOps(identifier);
-      TableMetadata lastMetadata = ops.current();
       glue.deleteTable(DeleteTableRequest.builder()
           .catalogId(awsProperties.glueCatalogId())
           .databaseName(IcebergToGlueConverter.getDatabaseName(identifier))
           .name(identifier.name())
           .build());
       LOG.info("Successfully dropped table {} from Glue", identifier);
+      if (purge && awsProperties.glueLakeFormationEnabled() && lastMetadata != null) {
```

Comment:
We can directly do the check upfront right? If `purge && awsProperties.glueLakeFormationEnabled()` we should directly throw error without loading last metadata.

Reply:
In that case we won't be able to drop table if `purge && awsProperties.glueLakeFormationEnabled()` even S3 data is already deleted.